app/src/main/java/com/google/android/ground/ui/datacollection/tasks/polygon/DrawAreaTaskFragment.kt

```diff
@@ -80,11 +85,31 @@ class DrawAreaTaskFragment @Inject constructor() : AbstractTaskFragment<DrawArea
     addUndoButton { removeLastVertex() }
     nextButton = addNextButton()
     addPointButton =
-      addButton(ButtonAction.ADD_POINT).setOnClickListener { viewModel.addLastVertex() }
+      addButton(ButtonAction.ADD_POINT).setOnClickListener {
+        viewModel.addLastVertex()
+        vibrate()
+      }
     completeButton =
       addButton(ButtonAction.COMPLETE).setOnClickListener { viewModel.completePolygon() }
   }
 
+  private fun vibrate() {
```

Comment:
Let's move the logic for vibration to a new system manager class and put it under the appropriate package. Also, let's call the vibrate function from within the view model and not fragment